tests/test_aggregates.py

```diff
@@ -263,50 +282,66 @@ def keyfunction(job):
+            aggregate_instance._create_AggregateStore(project)
 
     def test_valid_select(self, setUp, project):
         def _select(job):
             return job.sp.i > 5
 
         aggregate_instance = aggregator.groupsof(1, select=_select)
-        aggregate_instance = aggregate_instance._create_AggregatesStore(project)
+        aggregate_store = aggregate_instance._create_AggregateStore(project)
         selected_jobs = []
         for job in project:
             if _select(job):
                 selected_jobs.append((job,))
-        assert list(aggregate_instance.values()) == selected_jobs
+        assert list(aggregate_store.values()) == selected_jobs
 
-    def test_storing_hashing(self, setUp, project, list_of_aggregates):
+    def test_store_hashing(self, setUp, project, list_of_aggregators):
         # Since we need to store groups on a per aggregate basis in the project,
         # we need to be sure that the aggregates are hashing and compared correctly.
         # This test ensures this feature.
-        def _create_storing(aggregator):
-            return aggregator._create_AggregatesStore(project)
-
-        list_of_storing = list(map(_create_storing, list_of_aggregates))
-        # The above list contains 14 distinct storing objects (because a storing
-        # object is differentiated on the basis of the `_is_aggregate` attribute of
-        # the aggregator). When this list is converted to set, then these objects are
-        # hashed first and then compared. Since sets don't carry duplicate values, we test
-        # whether the length of the set obtained from the list is equal to 14 or not.
-        assert len(set(list_of_storing)) == 14
-
-    def test_get_by_id(self, setUp, project, list_of_aggregates):
-        def _create_storing(aggregator):
-            return aggregator._create_AggregatesStore(project)
-
-        # The below set contains the default as well as manu non-default
-        # objects which stores aggregates.
-        list_of_storing = set(map(_create_storing, list_of_aggregates))
-        for stored_aggregate in list_of_storing:
-            for aggregate in stored_aggregate.values():
-                assert aggregate == stored_aggregate[get_aggregate_id(aggregate)]
+        list_of_stores = [
+            aggregator._create_AggregateStore(project)
+            for aggregator in list_of_aggregators
+        ]
+        assert len(list_of_stores) == len(list_of_aggregators)
+        # The above list contains 14 distinct store objects (because a
+        # store object is differentiated on the basis of the
+        # ``_is_default_aggregate`` attribute of the aggregator). When this
+        # list is converted to a set, then these objects are hashed first and
+        # then compared. Since sets don't carry duplicate values, we test
+        # whether the length of the set obtained from the list is equal to 14
+        # or not.
+        assert len(set(list_of_stores)) == 14
```

Comment:
I'm confused why we need both this test and the one below. The one above the comment seems better since it's robust to the fixture length changing.

Reply:
These are testing different things: all default aggregators are equivalent, so `len(set(list_of_stores)) != len(list_of_stores)`. The `set` removes some aggregators that have different parameters provided to the constructor but that are all equivalent (default aggregators). We can't easily make this test robust to the fixture length changing (only something like `len(fixtures) - num_known_duplicates`, which I don't think is helpful).

Reply:
I missed the `set` call, thanks for the clarification.